File: solutions/java/src/restaurantmanagementsystem/state/ReadyForPickupState.java
package restaurantmanagementsystem.state;

import restaurantmanagementsystem.model.OrderItem;

public class ReadyForPickupState implements OrderItemState {
    @Override
    public void next(OrderItem item) {
        // This is the key state. When it transitions, it notifies observers.
        item.notifyObservers();
    }

    @Override
    public void prev(OrderItem item) {
        item.setState(new PreparingState());
    }

    @Override
    public String getStatus() { return "READY_FOR_PICKUP"; }
}

File: solutions/csharp/restaurantmanagementsystem/States/ReadyForPickupState.cs
class ReadyForPickupState : IOrderItemState
{
    public void Next(OrderItem item)
    {
        item.NotifyObservers();
    }

    public void Prev(OrderItem item)
    {
        item.SetState(new PreparingState());
    }

    public string GetStatus() => "READY_FOR_PICKUP";
}